core/src/main/scala/org/ensime/util/FileUtil.scala

```diff
@@ -33,7 +33,11 @@ object RichFileObject {
 object FileUtils {
 
   def isScalaSourceFile(f: File): Boolean = {
-    f.exists && f.getName.endsWith(".scala")
+    f.getName.endsWith(".scala")
```

Comment:
IMO it's surprising that this would hit the disk. Probably a (my) mistake in the first place. Only one client of this method and it looks safe to change.